api/handlers/dataexplorerhandler.py

```diff
@@ -496,11 +508,20 @@ def search_size(self, return_type):
             }
         }
 
-        size = config.es.search(
-            index='data_explorer',
-            doc_type='flywheel',
-            body=body)['aggregations']['count']['value']
-        size = int(size*1.02)
+        if filters:
+            body["query"] = {
+                "bool": {
+                    "filter": filters
+                }
+            }
+        try:
+            size = config.es.search(
+                index='data_explorer',
+                doc_type='flywheel',
+                body=body)['aggregations']['count']['value']
+            size = int(size*1.02)
+        except TransportError:
+            self.abort(400, "Reuest to large, filter your request")
```

Comment:
A couple things:
1) This wouldn't be the failure point for large requests - it would just return 1 aggregation with a large count (since `size` is set to 0). 
2) A user could request more than 10,000 in their `size` param as well, so I would check size in `_parse_request` after resolving the size param. 
3) I would modify the error to say: 
> Request would return more than 10,000 results. Please add additional filters.